.github/workflows/release-branches-check.yml

```diff
@@ -0,0 +1,53 @@
+name: Release branches check
+
+on:
+  pull_request:
+    types: [opened]
+  push:
+    branches:
+      - juan-fernandez/add-job-to-check-release-branches
+
+jobs:
+  check-release-branch-v2:
+    if: "!contains(github.event.pull_request.labels.*.name, 'dont-land-on-v2.x')"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Squash current branch commits
```

Comment:
Any way to avoid the squash? Might be nice to know exactly which commits are conflicting.

Reply:
we squash PRs so this is the workflow we'd follow when doing it manually. 

I guess we could `git merge $BRANCH` from the release branch instead of `cherry-pick`ing a single commit. That'd give us the conflicting commit

Reply:
I guess you're right, we do squash and it's possible that some commit that would conflict no longer conflicts once squashed, especially if more commits are added to fix the conflict. Squashing is fine then 👍